contracts/staking/GoodFundManager.sol

```diff
@@ -285,17 +286,16 @@ contract GoodFundManager is DAOContract {
     /**
      * @dev Mint to users reward tokens which they earned by staking contract
      * @dev _user user to get rewards
-     * @dev _staking staking contract that has implementation of staking reward system
      */
      function mintReward(
-        address _user,
-        StakingContract _staking 
+        address _user
 
      ) public {
         
-        (uint blockReward,uint startBlock,uint endBlock, bool isBlackListed) = getStakingReward(address(_staking));
-        uint amount = _staking._mint(_user);
-        if(amount > 0 && isBlackListed == false){
+        Reward memory staking = rewardsForStakingContract[address(msg.sender)];
+        uint amount = StakingContract(address(msg.sender))._mint(_user);
+        require(staking.isInitialized == true , "Staking contracts reward has not initiliazed");
```

Comment:
- this should be the first line of code
- isInitialized is redundant. its enough to test that startBlock>0